### PR TITLE
Add support for logr

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

We have libraries like k8s.io/utils that need logging, we should have a way to use the `logr` in these scenarios. For example see https://github.com/kubernetes/utils/pull/70

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Now supports github.com/go-logr/logr based interfaces backed by klog
```

### CODE REVIEW

**Comment by neolit123** [comment]
/assign @munnerz 
FYI

File: klogr/klogr.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"encoding/json"
+	"fmt"
+	"runtime"
+	"sort"
+
+	"github.com/go-logr/logr"
+	"k8s.io/klog"
+)
+
+// New returns a logr.Logger which is implemented by glog.
```
Comment (thockin) [inline klogr/klogr.go]: s/glog/klog

File: klogr/klogr.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	klog.ErrorDepth(framesToCaller(), l.prefix, " ", msgStr, " ", errStr, " ", fixedStr, " ", userStr)
+}
+
+func (l klogger) V(level int) logr.InfoLogger {
+	new := l.clone()
+	new.level = level
+	return new
+}
+
+// WithName returns a new logr.Logger with the specified name appended.  glogr
```
Comment (thockin) [inline klogr/klogr.go]: s/glog/klog

**Comment by thockin** [comment]
Do we need the hold?

**Pull request merged**